File: backend/app/schemas/juice.py
"""烟油请求/响应模型"""
from pydantic import BaseModel


class JuiceCreate(BaseModel):
    """创建烟油请求"""
    brand_id: int
    name: str
    flavor_profile: str | None = None
    nicotine_range: str | None = None
    vg_pg_ratio: str | None = None
    volume: str | None = None
    price_range: str | None = None
    description: str | None = None
    image_urls: list[str] = []
    tag_ids: list[int] = []


class JuiceUpdate(BaseModel):
    """更新烟油请求"""
    brand_id: int | None = None
    name: str | None = None
    flavor_profile: str | None = None
    nicotine_range: str | None = None
    vg_pg_ratio: str | None = None
    volume: str | None = None
    price_range: str | None = None
    description: str | None = None
    image_urls: list[str] | None = None
    tag_ids: list[int] | None = None

File: frontend/src/api/juice.ts
import client from './client';

export interface TagItem {
  id: number;
  name: string;
}

export interface JuiceItem {
  id: number;
  brand_id: number;
  brand_name: string;
  name: string;
  flavor_profile: string | null;
  nicotine_range: string | null;
  vg_pg_ratio: string | null;
  volume: string | null;
  price_range: string | null;
  description: string | null;
  image_urls: string[];
  status: string;
  avg_rating: number;
  review_count: number;
  tags: TagItem[];
  created_at: string;
}

export interface JuiceListData {
  items: JuiceItem[];
  total: number;
  page: number;
  size: number;
}

export const getJuices = (params: {
  page?: number;
  size?: number;
  brand_id?: number;
  flavor_profile?: string;
  sort?: string;
}) => client.get<{ code: number; data: JuiceListData }>('/juices', { params });

export const getJuice = (id: number) =>
  client.get<{ code: number; data: JuiceItem }>(`/juices/${id}`);

export const searchJuices = (q: string, page = 1, size = 20) =>
  client.get<{ code: number; data: JuiceListData }>('/juices/search', { params: { q, page, size } });

export const getTopRated = (limit = 20) =>
  client.get<{ code: number; data: JuiceItem[] }>('/juices/top-rated', { params: { limit } });
